Mathlib/Topology/Algebra/Module/Dual.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+import Mathlib.LinearAlgebra.SesquilinearForm
+import Mathlib.Topology.Algebra.Module.Basic
+
+/-!
+# Topological dual
```

Comment:
I think the module docstring of this new file should briefly explain the meaning of the concept of dual (however standard) as well as the notation `->L[k]` (which is not standard in mathematics, only standard in parts of Mathlib). The docstring of the file [`NormedSpace.Dual`](https://github.com/leanprover-community/mathlib4/blob/f2eace05e038fd0f6da83e3604db037040d89d54/Mathlib/Analysis/NormedSpace/Dual.lean) contained an appropriate amount of detail in my opinion. Copying the authorship from there would also seem appropriate at least unless the contents substantially change or are expanded.

Reply:
I wrote a bit in the module docstring. I added you to the list of authors as well since I took the content from the `WeakDual`-file